src/main/scala/com/daml/projection/scaladsl/BatchSource.scala

```diff
@@ -30,8 +31,13 @@ trait BatchSource[E] {
 }
 
 object BatchSource {
+  trait EventForFilter[E] {
```

Comment:
This needs doc

Reply:
Spin off task for GetContractTypeId and GetParties in both scaladsl and javadsl